logic/block.js

```diff
@@ -199,10 +200,11 @@ class Block {
 				.createHash('sha256')
 				.update(dataWithoutSignature)
 				.digest();
-			const blockSignatureBuffer = Buffer.from(block.blockSignature, 'hex');
-			const generatorPublicKeyBuffer = Buffer.from(
-				block.generatorPublicKey,
-				'hex'
+			const blockSignatureBuffer = this.scope.ed.hexToBuffer(
+				block.blockSignature
+			);
+			const generatorPublicKeyBuffer = this.scope.ed.hexToBuffer(
+				block.generatorPublicKey
```

Comment:
So you are using `ed` from scope, not the one declared at top level. So better not to require it.